File: src/node/table.js
import "colors";
import { Gestell } from "@gestell/sdk";

const gestell = new Gestell();

async function start() {
  // Retrieve the Gestell Guide Collection
  const collections = await gestell.collection.list({
    search: "Gestell Guide Collection",
  });

  if (collections.result.length === 0) {
    console.log(
      "The Gestell Guide Collection has not been created, run setup.js first"
        .yellow.bold
    );
    return;
  }

  const collectionId = collections.result[0].id;
  const collection = await gestell.collection.get(collectionId);
  console.log(collection);

  const categories = collection.result.categories;
  let categoryId = "";

  for (const category of categories) {
    if (category.type === "table") {
      categoryId = category.id;
    }
  }

  if (!categoryId) {
    console.log(
      "The Gestell Guide Collection does not have the table category".yellow
        .bold
    );
    return;
  }

  const response = await gestell.query.table({
    collectionId,
    categoryId,
  });
  for (const item of response.result) {
    console.log(item);
  }
}

(async () => start())();

File: src/python/table.py
import asyncio
from gestell import Gestell

gestell = Gestell()


async def start():
    # Retrieve the Gestell Guide Collection
    collections = await gestell.collection.list(search="Gestell Guide Collection")

    if not collections.result:
        print(
            "\033[93m\033[1mThe Gestell Guide Collection has not been created, run setup.js first\033[0m"
        )
        return

    collection_id = collections.result[0].id
    collection = await gestell.collection.get(collection_id)

    categories = collection.result.categories
    category_id = ""

    for category in categories:
        if category.type == "table":
            category_id = category.id

    if not category_id:
        print(
            "\033[93m\033[1mThe Gestell Guide Collection does not have the table category\033[0m"
        )
        return

    response = await gestell.query.table(
        collection_id=collection_id,
        category_id=category_id,
    )
    for item in response.result:
        print(item)


if __name__ == "__main__":
    asyncio.run(start())
